Bot/Storage.py

```diff
@@ -1,30 +1,55 @@
-import datetime
+from datetime import datetime
+import json
+import os
 from typing import Optional
-
+from logging import Logger
 
 class Storage:
-    def __init__(self) -> None:
+    def __init__(self, logger : Logger) -> None:
         """Stores the information for each user"""
         # Load the shelve db if possible
-        self.storage = {}
+        if os.path.isfile(f"./database/storage.json"):
+            logger.info("Loading stored events...")
+            with open("./database/storage.json", "r") as storageFile:
+                try:
+                    self.storage = json.load(storageFile)
+                except Exception as e:
+                    logger.error("Error loading stored events", {str(e)})
+                    self.storage = {}
+        else:
+            logger.info("No saved events found.")
+            self.storage = {}
+            os.makedirs("./database", exist_ok=True)
+            with open("./database/storage.json", "w") as storageFile:
+                json.dump(self.storage, storageFile, indent=4)
+            
 
-    def add_event(self, chat_id: int, event_name: str, event_time: str) -> datetime.datetime:
+    def add_event(self, chat_id: int, event_name: str, event_time: str) -> datetime:
         """
         Throws ValueError when the format is incorrect
         """
+        chat_id = str(chat_id)
```

Comment:
Maybe we can create another class to deal with the file logging (opening/closing/writing to files ) and only write to the file when the destructor of the class is called.